### PR TITLE
fix(Resource Manager): update service after recent API changes and generate examples

### PR DESCRIPTION
**Pull request by Andras-Csanyi**
## PR summary
This PR updates the Resource Manager service, adds examples and move configuration values out from tests into a configuration file.

## PR Checklist
Please make sure that your PR fulfills the following requirements:  
- [x] The commit message follows the [Angular Commit Message Guidelines](https://github.com/angular/angular/blob/master/CONTRIBUTING.md#-commit-message-guidelines).
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## Does this PR introduce a breaking change?    
- [ ] Yes
- [x] No

### CODE REVIEW

File: examples/resource-manager.v2.test.js [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+// RESOURCE_MANAGER_APIKEY=<IAM apikey>
+// RESOURCE_MANAGER_AUTH_URL=<IAM token service base URL - omit this if using the production environment>
+// RESOURCE_MANAGER_TEST_QUOTA_ID=<Quota Id used in testing>
+// RESOURCE_MANAGER_TEST_USER_ACCOUNT_ID=<User Id used in testing>
+//
+// These configuration properties can be exported as environment variables, or stored
+// in a configuration file and then:
+// export IBM_CREDENTIALS_FILE=<name of configuration file>
+//
+const configFile = 'resource_manager_v2.env';
```
Comment (pyrooka) [inline examples/resource-manager.v2.test.js:40]: Just simply `resouce_manager.env`

File: examples/resource-manager.v2.test.js [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+  
+  // begin-common
+  
+  const resourceManagerService = ResourceManagerV2.newInstance({});
+  
+  // end-common
+  
+  const config = readExternalSources(ResourceManagerV2.DEFAULT_SERVICE_NAME);
+  const testUserAccountId = config.testUserAccountId;
+  const testQuotaId = config.testQuotaId;
```
Comment (pyrooka) [inline examples/resource-manager.v2.test.js:62]: We are using these 2 params without the test prefix in the config file.

File: test/integration/resource-manager.v2.test.js [hunk 3]
```diff
@@ -30,25 +31,47 @@ const describe = authHelper.prepareTests(configFile);
...
+  let resourceManagerUsersService = null;
+  let newResourceGroupId = null;
+  let url = null;
+  let authType = null;
+  let apiKey = null;
+  let authUrl = null;
+  let testQuotaId = null;
+  let testUserAccountId = null;
+
+  beforeAll('should successfully complete initialization', done => {
```
Comment (pyrooka) [inline test/integration/resource-manager.v2.test.js:44]: The first argument is a callback function, not a string. Check the other integration tests (search for `beforeAll` in the project)

File: test/integration/resource-manager.v2.test.js [hunk 4]
```diff
@@ -30,25 +31,47 @@ const describe = authHelper.prepareTests(configFile);
...
+    resourceManagerService = ResourceManagerV2.newInstance({ serviceName: 'RMGR1' });
```
Comment (pyrooka) [inline test/integration/resource-manager.v2.test.js:45]: We should use the default service name constant here.

File: test/integration/resource-manager.v2.test.js [hunk 5]
```diff
@@ -30,25 +31,47 @@ const describe = authHelper.prepareTests(configFile);
...
+    expect(resourceManagerService).not.toBeNull();
+    resourceManagerUsersService = ResourceManagerV2.newInstance({ serviceName: 'RMGR2' });
```
Comment (pyrooka) [inline test/integration/resource-manager.v2.test.js:47]: Please rename the variable to `deleteResourceManagerService` and change the service name to `ALT_RESOURCE_MANAGER`.

File: test/integration/resource-manager.v2.test.js [hunk 6]
```diff
@@ -30,25 +31,47 @@ const describe = authHelper.prepareTests(configFile);
...
+    expect(resourceManagerUsersService).not.toBeNull();
+
+    const config = readExternalSources(ResourceManagerV2.DEFAULT_SERVICE_NAME);
+    expect(config).not.toBeNull();
+    url = config.url;
+    authType = config.authType;
+    apiKey = config.apiKey;
+    authUrl = config.AuthUrl;
```
Comment (pyrooka) [inline test/integration/resource-manager.v2.test.js:55]: `config.authUrl`

File: test/integration/resource-manager.v2.test.js [hunk 7]
```diff
@@ -30,25 +31,47 @@ const describe = authHelper.prepareTests(configFile);
...
+    testQuotaId = config.testQuotaId;
+    testUserAccountId = config.testUserAccountId;
```
Comment (pyrooka) [inline test/integration/resource-manager.v2.test.js:57]: We are using these 2 config parameters without the test prefix in the env file.

**Review by pyrooka — CHANGES_REQUESTED** [review]

File: examples/resource-manager.v2.test.js [hunk 8]
```diff
@@ -0,0 +1,248 @@
...
+
+// Mocks for console.log and console.warn
+const consoleLogMock = jest.spyOn(console, 'log');
+const consoleWarnMock = jest.spyOn(console, 'warn');
+
+describe('ResourceManagerV2', () => {
+  
+  // begin-common
+  
+  const resourceManagerService = ResourceManagerV2.newInstance({});
```
Comment (pyrooka) [inline examples/resource-manager.v2.test.js:56]: We should create another service instance for the delete operation. Let's call it `deleteResourceManagerService` and the use the service name `ALT_RESOURCE_MANAGER`.

File: examples/resource-manager.v2.test.js [hunk 9]
```diff
@@ -0,0 +1,248 @@
...
+      done(output);
+    });
+    
+    // begin-delete_resource_group
+    
+    const params = {
+      id: resourceGroupId,
+    };
+    
+    resourceManagerService.deleteResourceGroup(params)
```
Comment (pyrooka) [inline examples/resource-manager.v2.test.js:188]: Use the newly created delete service here.

**Review by pyrooka — CHANGES_REQUESTED** [review]

File: examples/resource-manager.v2.test.js [hunk 10]
```diff
@@ -0,0 +1,257 @@
...
+      originalLog(output);
+      done();
+    });
+    consoleWarnMock.mockImplementation(output => {
+      done(output);
+    });
+    
+    // begin-list_resource_groups
+    const params = {
+      accountId: userAccountId
```
Comment (padamstx) [inline examples/resource-manager.v2.test.js:168]: add includeDeleted true

**Review by padamstx — CHANGES_REQUESTED** [review]
Looks good, but I suggested a few minor changes.

**Review by padamstx — APPROVED** [review]
LGTM

**Pull request merged**